app/src/main/res/layout/activity_login.xml

```diff
@@ -42,12 +42,60 @@
                     android:scaleType="fitXY" />
 
                 <android.support.design.widget.TextInputLayout
-                    android:id="@+id/username_layout"
+                    android:id="@+id/login_process_layout"
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
                     android:layout_below="@id/icon_iv"
                     android:layout_centerInParent="true"
                     android:textColor="@color/colorPrimaryDark"
+                    android:visibility="@{VIEW.VISIBLE}">
+
+                    <android.support.v7.widget.AppCompatEditText
+                        android:id="@+id/generic_login_error_tv"
+                        android:layout_width="300dp"
+                        android:layout_height="0dp"
+                        android:inputType="text" />
+                </android.support.design.widget.TextInputLayout>
+
+                <android.support.design.widget.TextInputLayout
+                    android:id="@+id/client_secret_layout"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_below="@id/login_process_layout"
+                    android:layout_centerInParent="true"
+                    android:textColor="@color/colorPrimaryDark"
+                    android:visibility="@{VIEW.VISIBLE}">
```

Comment:
I don't think these should be displayed. At some the app should retrieve these automatically. I don't really see why the user should input them by hand.